app/views/kudos/index.html.erb

```diff
@@ -14,13 +14,13 @@
 
   <tbody>
     <% @kudos.each do |kudo| %>
-      <tr>
+      <tr data-kudo-id="<%= kudo.id %>">
         <td><%= kudo.title %></td>
         <td><%= kudo.content %></td>
         <td><%= kudo.giver.email %></td>
         <td><%= kudo.receiver.email %></td>
-        <td><%= kudo.company_value.title %></td>
-        <% if kudo.giver == current_employee %>
+        <td><%= kudo.company_value.title %>        </td>
+        <% if kudo.giver == current_employee && policy(kudo).update? %>
```

Comment:
I think it should be enough, since you are checking current user in pundit anyway

```suggestion
        <% if policy(kudo).update? %>
```